pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -893,10 +895,14 @@ func (cr *PerconaXtraDBCluster) ProxySQLEnabled() bool {
 	return cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Enabled
 }
 
-func (s *PerconaXtraDBClusterStatus) ClusterStatus(inProgress bool) AppState {
+func (s *PerconaXtraDBClusterStatus) ClusterStatus(inProgress, deleted bool) AppState {
 	switch {
 	case s.PXC.Status == AppStateError || s.ProxySQL.Status == AppStateError || s.HAProxy.Status == AppStateError:
 		return AppStateError
+	case deleted || s.PXC.Status == AppStateStopping || s.ProxySQL.Status == AppStateStopping || s.HAProxy.Status == AppStateStopping:
+		return AppStateStopping
+	case s.PXC.Status == AppStatePaused && (s.ProxySQL.Status == AppStatePaused || s.HAProxy.Status == AppStatePaused):
```

Comment:
what if both ProxySQL and HAProxy are disabled?

Reply:
fixed in https://github.com/percona/percona-xtradb-cluster-operator/pull/894/commits/650a935ad7d1abf5850c129ac42584562e06c069